whereby/browser-sdk
# Auto switch stream when device is unplugged

### Description

Sorry for the big PR on this small bugfix, but we didn't really handle switching of streams at all, so I had to add a bunch of logic. Also did some refactoring.

**Summary:**

* Handle unplugging of external devices (Auto-switch stream to next device)
* Handle busy devices, store a list of them in the state, and don't try to switch to a busy device
* Refactor local media slice. Aligning it with the rest of the slices, action creator naming, reducer structure etc
* Add a new slice for local screenshare. Moving the screenshare stuff out of local media and in to a dedicated slice.

**Related Issue:**



### Testing

1. Test with storybook.
2. Verify that all local media functionality works as before
3. Change the camera device to an external one
4. Unplug it
5. Verify that the stream switches to the next device in the list
6. Plug the camera back
7. Verify that the stream does not switch automatically, but that the camera becomes available for manual switching

### Screenshots/GIFs (if applicable)



### Checklist

        under the terms of the MIT license.

### Dependency Updates



### Reviewers



@havardholvik
@kevinwhereby
@nandito
@thyal

### Additional Information
